File: articles/inequality-aversion-economic-decision-making/go/inequality_aversion_utility.go
package main

import (
	"fmt"
	"math"
)

func fehrSchmidt(selfPayoff, otherPayoff, alpha, beta float64) float64 {
	return selfPayoff -
		alpha*math.Max(otherPayoff-selfPayoff, 0.0) -
		beta*math.Max(selfPayoff-otherPayoff, 0.0)
}

func main() {
	fmt.Printf("Fehr-Schmidt utility: %.3f\n", fehrSchmidt(0.30, 0.70, 1.5, 0.6))
}

File: articles/inequality-aversion-economic-decision-making/rust/inequality_aversion_utility.rs
fn fehr_schmidt(self_payoff: f64, other_payoff: f64, alpha: f64, beta: f64) -> f64 {
    self_payoff
        - alpha * (other_payoff - self_payoff).max(0.0)
        - beta * (self_payoff - other_payoff).max(0.0)
}

fn main() {
    println!("Fehr-Schmidt utility: {:.3}", fehr_schmidt(0.30, 0.70, 1.5, 0.6));
}
